### ISSUE TITLE
draft-path parameters

### ISSUE THREAD

**Issue by nottwy** [HingeAssembler/HINGE#129]
Dear developer,

I want to make sure some parameters that module 'draft-path' uses.

hinge draft-path $PWD $db hsy50x.G2.graphml
1. I use the name of daligner database here. Is it correct?
2. should I use G2 or G3?

In your manual, you write G2. But in a issue,'https://github.com/HingeAssembler/HINGE/issues/77', you write G3 there. Is there any difference?

thank you.

**Comment by govinda-kamath** [comment]
Yes. The name of the daligner database as `$db`.  And you should use G2. `G3` is still an experimental module for larger genomes.

**Comment by nottwy** [comment]
@govinda-kamath ,
I'm assembling a large genome. And I don't see G3 file.
I run the pipeline provided in issue 'https://github.com/HingeAssembler/HINGE/issues/77', but I don't see the G3 file. Is it caused by a wrong operation?

**Comment by govinda-kamath** [comment]
Which commit of the software are you running?

**Comment by nottwy** [comment]
I install it in Jun 28,2017 with git clone.
Which file should look to see the commit?

**Comment by govinda-kamath** [comment]
```
git log
```
should give you the commit that you installed.

But it's weird that there is no G3.

**Comment by nottwy** [comment]
The first few rows look like this:
commit 459e3096a4bd0b6d41c6ca153dd8a2e74e7a14ad
Author: Govinda Kamath <govinda.kamath@gmail.com>
Date:   Wed Jun 21 16:20:54 2017 +0530

**Comment by govinda-kamath** [comment]
This should actually still be producing a `G3` file. Can you confirm the size of the `G2` file?

**Comment by nottwy** [comment]
464K Sep 12 14:29 hsy50x.G2c.graphml
418K Sep 12 14:29 hsy50x.G2s.graphml
457K Sep 12 14:29 hsy50x.Gc.graphml
411K Sep 12 14:29 hsy50x.Gs.graphml
756K Sep 12 14:29 hsy50x.G1.graphml
756K Sep 12 14:29 hsy50x.G2.graphml
-rw-rw-r-- 1 wxtao wxtao 4.5M Sep 12 14:29 hsy50x.G0.graphml
-rw-rw-r-- 1 wxtao wxtao 6.7M Sep 12 14:28 hsy50x.G00.graphml

**Comment by govinda-kamath** [comment]
It looks like something went wrong in the `draft-path` run. `G0`, `G1`, `G2`, `G3` are usually of similar sizes.

Can you return the STDOUT of the run of the `draft-path`?

**Comment by nottwy** [comment]
**command:**
$hinge clip hsy.edges.hinges hsy.hinge.list 50x

**output:**
Tue Sep 12 16:49:22 CST 2017
0 bad coverage reads.
0 bad self aligned reads.
Tue Sep 12 16:50:43 CST 2017

**Comment by govinda-kamath** [comment]
Can you give us the `hsy.edges.hinges` and `hsy.hinge.list` files? These files contain no sequence information, in case you're worried about privacy.

**Comment by nottwy** [comment]
I can provide it to you.
But when I prepare the data, I made a mistak and deleted 'hsy.hinge.list'.
I know you want to rerun in your local machine so it's useless if I only provide you 'hsy.edges.hinges'.

It must be a mistake of your program. And now I want to know, is it ok if I use G2 file?

**Comment by govinda-kamath** [comment]
Sure. You can. Though it looks like the code crashed somewhere there (so I'm not sure about if the results will be kosher).

**Comment by nottwy** [comment]
it needs one or two day to get the file 'hsy.hinge.list'.
And I will mail it to you after I rerun the program. Can you provide me your email?

**Comment by nottwy** [comment]
Another question, do you think HINGE is suited for a large genome with relative high repeat content?

**Comment by govinda-kamath** [comment]
Please send it to gkamath@stanford.edu and feixia@stanford.edu.

**Comment by govinda-kamath** [comment]
How large a genome are you interested in?

**Comment by nottwy** [comment]
~1 Gb

**Comment by govinda-kamath** [comment]
Hinge should work for a 1 GB genome.

**Comment by nottwy** [comment]
Have you received my email?

**Comment by govinda-kamath** [comment]
Yes. We'll get back to you soon.

**Comment by ilanshom** [comment]
There were a few small issues with hinge clip that were preventing it from creating the G3 graph. We created a new branch called devG3, which should fix that. 

In order to have the G3 graph produced, you should make sure that `aggressive_pruning = true` is in your nominal.ini file (we added it in devG3), and you should call `hinge clip` with the path to the ini file as the fourth argument. So your command should look like:

```hinge clip hsy.edges.hinges hsy.hinge.list test path-to-ini/nominal.ini``` 

Let us know if this works.

**Comment by nottwy** [comment]
Before I report my result I want to say another thing: Your tool is really difficult to compile. I'm serious.
And now let's return to our topic. I haven't installed the devG3 branch successfully yet. I try to run the clip step with the old version of hinge. And it gives me an error message like this:
bad coverage reads.
0 bad self aligned reads.
couldn't finish sparsification10328
couldn't finish sparsification10382

I will try to install the devG3 branch and try again. And I hope this error message can help you solve other problems of hinge. I'll report the progess of devG3 in this issue later.

**Comment by govinda-kamath** [comment]
These are just reports on the graph about visualisation, not error messages. You should be able to continue on the pipeline.
